hypothesis-python/src/hypothesis/errors.py

```diff
@@ -23,6 +23,9 @@ class UnsatisfiedAssumption(HypothesisException):
     If you're seeing this error something has gone wrong.
     """
 
+    def __init__(self, *, reason=None):
+        self.reason = reason
```

Comment:
Apparently we have a test for whether we have a test for all error classes with custom init methods!

Reply:
`test_exceptions_are_picklable` actually caught a bug due to this, which is that `UnsatisfiedAssumption` was no longer picklable after making `reason` a kwarg. To be honest, `reason` seems sensible as a posarg instead, so I turned it into one (which consequently fixed this issue).

I'm not particularly familiar with pickling issues like this, so maybe there is a more obvious solution.